numpy/lib/function_base.py

```diff
@@ -1320,6 +1321,214 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def _inplace_sum(items): 
+    """
+    Returns the sum of the elements of a list. 
+    Used in the follwing functions:
+    divergence,
+
+        .. versionadded:: 1.11.0
+
+    """
+    it = iter(items)
+    total = next(it)
+    for item in it:
+        total += item
+    return total
```

Comment:
I think this can be written `np.sum(arr, out=arr[0])`. Seems like a premature optimization to me though